python/ray/serve/tests/test_gradio.py

```diff
@@ -28,9 +29,21 @@ def test_gradio_ingress_correctness(serve_start_shutdown):
     def greet(name):
         return f"Good morning {name}!"
 
-    app = GradioServer.bind(
-        lambda: gr.Interface(fn=greet, inputs="text", outputs="text")
-    )
+    if use_user_defined_class:
+
+        @serve.deployment
+        class UserDefinedGradioServer(GradioIngress):
+            def __init__(self):
+                super().__init__(
+                    lambda: gr.Interface(fn=greet, inputs="text", outputs="text")
+                )
```

Comment:
@zcin I added this test case because we didn't have any tests for this use case. In the previous revision I had actually broken this API because I made `__init__` `async def`. Fixed now and added a test.

In the future we need to make sure to have thorough coverage of any such public API to avoid breakages.

Reply:
Ack, thanks for catching and fixing this one